extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -292,6 +295,7 @@ class CornerstoneViewportService implements IViewportService {
     }
 
     viewport.setStack(imageIds, initialImageIndexToUse).then(() => {
+      hangingProtocolService.applyCustomViewportSettings(viewportOptions, viewport, properties);
```

Comment:
General comment (thinking out loud)

First of all we should be consistent between settings and options. If it is viewportOptions, then it should be `applyCustomViewportOptions`

I understand what you are doing here. However, I don't understand why we are adding `viewportSettings/Options` through the `HangingProtocolService` and then calling the `apply...`. Can't we just those callbacks here, instead of adding and then applying?

`customViewportOptions` was originally added to have simple key-value pairs, but I also understand the value of the functions, which makes `Options` more general. I think `customViewportOptions` should remain as simple key-value pairs accessible throughout the code and maybe we add a new `customViewportCallbacks` that get executed? but can we serialize that in JSON?

Reply:
I also don't understand why `properties` is being passed to the apply callback?

Reply:
Changed the name consistently to applyCustomViewportOptions
The reason one adds them through the HangingProtocolService is that they are defined in the hanging protocol service as part of hte protocol itself - that is, in 
export type PublicViewportOptions = {
  customViewportOptions?: Record<string, unknown>;
};
it already defines the key/value pairs.  However, it doesn't define what to do with them, and if they can be handled by the extension/viewport directly, then they should be direct options - you didn't like adding direct options to CS3D, so one needs the ability to apply the custom options in some fashion.  That pretty much requires a setter.  Now, it would be possible to define that on the cornerstone service itself, but it makes sense to put them with the spot they are defined, thus on the hanging protocol service.  It doesn't matter what type of viewport it is - if the viewport is a completely new one, then one just needs to write new apply code for it.
It doesn't make sense to add generic custom viewport options for all hanging protocols, so I think this design is in line with the general design.

Reply:
properties is just being passed because I thought it might be important to have access to.  I'm not currently using it, but I wasn't sure if it would be required.  There is no guarantee as to what gets passed beyond the viewport